src/arm/arm.c

```diff
@@ -87,6 +87,8 @@ mraa_arm_platform()
             platform_type = MRAA_96BOARDS;
         else if (mraa_file_contains("/proc/device-tree/model", "s900"))
             platform_type = MRAA_96BOARDS;
+        else if (mraa_file_contains("/sys/firmware/devicetree/base/compatible", "raspberrypi,"))
```

Comment:
`/sys/firmware/devicetree/base` should be `/proc/device-tree` to align with the kernel recommendation and the very change one of the other patches in this PR introduces.